### PR TITLE
[terra-html-table] Update docs to use mdx syntax

### PR DESCRIPTION
**Pull request by pranav300**
### Summary
updated docs to use MDX syntax

### CODE REVIEW

File: packages/terra-html-table/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 Changelog
 =========
+### Changed
+* Update docs to use mdx syntax.
```
Comment (emilyrohrbough) [inline packages/terra-html-table/CHANGELOG.md]: whoops, this should be under the unreleased section

Comment (pranav300) [reply packages/terra-html-table/CHANGELOG.md]: done here https://github.com/cerner/terra-core/pull/2850/commits/5db8bce68a204971a17a3039ecf58b9849ca0e1e.

File: packages/terra-html-table/src/terra-dev-site/doc/html-table/Table.1.doc.mdx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+## Peer Dependencies
+
+This component requires the following peer dependencies be installed in your app for the component to properly function.
+
+| Peer Dependency | Version |
+|-|-|
+| react | ^16.8.5 |
+| react-dom | ^16.8.5 |
+
+<!-- AUTO-GENERATED-CONTENT:END -->
```
Comment (emilyrohrbough) [inline packages/terra-html-table/src/terra-dev-site/doc/html-table/Table.1.doc.mdx]: The usage section is missing for the quick reference to the component import(s)

Comment (pranav300) [reply packages/terra-html-table/src/terra-dev-site/doc/html-table/Table.1.doc.mdx]: updated here https://github.com/cerner/terra-core/pull/2850/commits/1c7c026fa2a75274daed84048871c0613d76d22a.

**Pull request merged**